conformance/tests/gatewayclass-observed-generation-bump.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, GatewayClassObservedGenerationBump)
+}
+
+var GatewayClassObservedGenerationBump = suite.ConformanceTest{
+	ShortName:   "GatewayClassObservedGenerationBump",
+	Description: "A GatewayClass should update the observedGeneration in all of it's Status.Conditions after an update to the spec",
```

Comment:
currently Contour is not reconciling changes to GatewayClasses, just taking their state when they are first created/observed by the Contour Gateway API provisioner due to:

https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.GatewayClass

```
It is recommended that this resource be used as a template for Gateways. This means that a Gateway is based on the state of the GatewayClass at the time it was created and changes to the GatewayClass or associated parameters are not propagated down to existing Gateways. This recommendation is intended to limit the blast radius of changes to GatewayClass or associated parameters. If implementations choose to propagate GatewayClass changes to existing Gateways, that MUST be clearly documented by the implementation.
```

We haven't yet finished our thinking/implementation on this area so this fails for us

Reply:
Given that `GatewayClass` has an `Accepted` condition I would expect the observedGeneration of that condition to increment when the GWC spec changes.

Reply:
ah nvm, found where we were short circuiting the status update logic, this should work with Contour we have a small fix to make

Reply:
https://github.com/projectcontour/contour/pull/4932